### DISCUSSION TITLE
Create tabix index from scratch

### DISCUSSION THREAD

**Discussion by holtgrewe** [zaeleus/noodles#208] [Q&A] [answered]
Hello, where can I find an example on how to create a tabix index from scratch? My use case is that I want to generate a `.tbi` file for a `.vcf.gz` file that I'm writing, ideally create the in memory data structures on the fly and then write it out.

A second use case would of course be read `.vcf.gz` and create the `.tbi` file from this.

**Comment by zaeleus** [comment, marked as answer]
The strategy for building an index is to track the start and end virtual positions of each record. Each of these pairs is called a chunk. For tabix, the reference sequence name, record start position, record end position, and chunk are then used by the indexer.

For your first use case, see [`tabix_write`](https://github.com/zaeleus/noodles/blob/2ae1c91543b5decb02e9e72befb5e71ad2830090/noodles-tabix/examples/tabix_write.rs). This shows writing and indexing a BED-like record structure, but it can be applied to VCF as well.

[`vcf_index`](https://github.com/zaeleus/noodles/blob/2ae1c91543b5decb02e9e72befb5e71ad2830090/noodles-vcf/examples/vcf_index.rs) shows an example of reading a bgzip-compressed VCF and writing a tabix index for it.

Please also update to noodles 0.54.0 / noodles-csi 0.26.0. While adding the `tabix_write` example, I discovered a bug (fixed in 599aefac86702ff878aed448ebdc1f7fad97ee70) in the builder, so thank you for asking about indexing!
